lib/ronin/db/dns_query.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+#
+# ronin-db-activerecord - ActiveRecord backend for the Ronin Database.
+#
+# Copyright (c) 2022-2023 Hal Brodigan (postmodern.mod3 at gmail.com)
+#
+# ronin-db-activerecord is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ronin-db-activerecord is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with ronin-db-activerecord.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+require 'ronin/db/model'
+
+require 'active_record'
+require 'resolv'
+
+module Ronin
+  module DB
+    #
+    # Represents a DNS query.
+    #
+    class DNSQuery < ActiveRecord::Base
+
+      include Model
+
+      # @!attribute [rw] id
+      #   The primary ID for the DNS query.
+      #
+      #   @return [Integer]
+      attribute :id, :integer
+
+      # @!attribute [rw] type
+      #   The queried record type.
+      #
+      #   @return [String]
+      enum type: {
+        A:     'A',
+        AAAA:  'AAAA',
+        ANY:   'ANY',
+        CNAME: 'CNAME',
+        HINFO: 'HINFO',
+        LOC:   'LOC',
+        MX:    'MX',
+        NS:    'NS',
+        PTR:   'PTR',
+        SOA:   'SOA',
+        SRV:   'SRV',
+        TXT:   'TXT',
+        WKS:   'WKS'
+      }
```

Comment:
Maybe also add `_suffix: :query` that way we get methods named `any_query?` and `txt_query?`, as opposed to just `a?`, `aaaa?`, `any?`.